inc/vcf/record_cache.hpp

```diff
@@ -105,6 +105,25 @@ namespace ebi
             list_duplicates.clear();
             list_symbolic_duplicates.clear();
 
+            if (current_chromosome != record.chromosome) {
+                // EVA-1950: the cache removes "lexicographically lower" chromosomes when the cache is full, but with
+                // that sorting, "chr10" is lower than "chr2", so new variants in "chr10" get deleted to keep the ones
+                // in "chr2", and duplicates in "chr10" will go undetected. Possible solutions (better first):
+                // 1. The chosen simple solution is to keep in the cache variants from only one chromosome.
```

Comment:
How will this solution scale with VCF with millions of contigs?

Reply:
it's unaffected by the number of contigs. This solution keeps the variants of one chromosome *at a time*, so when the next one comes, the cache is cleared and the next chr can fill the cache. I doesn't matter how many chrs/contigs there are. 
The only hypothetical case that this cache is not enough by default is if there are  999 variants between 2 duplicate variants. In that case the 2 duplicate variants would pass undetected. I can give some artificial examples that would trigger that, but they look quite unrealistic.